plugin/storage/factory.go

```diff
@@ -201,6 +208,14 @@ func (f *Factory) InitFromViper(v *viper.Viper) {
 }
 
 func (f *Factory) initDownsamplingFromViper(v *viper.Viper) {
+	// if the downsampling flag isn't set then this component used the standard "AddFlags" method
+	// and has no use for downsampling.  the default settings effectively disable downsampling
+	if !v.IsSet(downsamplingRatio) {
```

Comment:
to clarify, `v.IsSet` does not mind if the flag was not even declared? I thought `IsSet` refers to whether the user set it or not.

Reply:
perhaps a cleaner alternative is to have a field in the factory that we set to true when addDownsamplingFlags is called

Reply:
I believe you're right about `IsSet`.  I don't like adding state here, but I've played with some of these other methods in viper (AllSettings, AllKeys) and they are more complex than they appear on the surface.  Agree the cleanest implementation here will be adding a bool